src/commands/app/deploy.ts

```
@@ -154,57 +153,81 @@ function downloadDeployLogImpl(
       }, 3000);
     });
 }
-
-async function createArtifact(context: CLIContext, checksum: Checksum) {
-  console.log(chalk`Uploading archive`);
-  const result = await controller.createArtifactUpload(context, checksum);
-  const stream = createArchiveReadStream();
-  await controller.uploadArtifact(result.uploadRequest, checksum.md5, stream);
-  console.log(chalk`Archive uploaded`);
-  const artifactID = await controller.createArtifact(
-    context,
-    result.artifactRequest
-  );
-  console.log(chalk`Artifact created`);
-  return artifactID;
-}
+*/
 
 async function run(argv: Arguments) {
-  const name = argv['cloud-code'] as string;
-  // TODO: support deploying all cloud code at once
-  // skygear-controller need an api to support batch deploy
-  if (name == null || name === '') {
-    console.error(chalk`Need name of cloud code, use --name`);
-    return;
+  const deploymentMap = argv.appConfig.deployments || {};
+  if (!Object.keys(deploymentMap).length) {
+    return Promise.reject('No deployment items to be deploy.');
```

Since this is a async function, it is better to just throw an error to make the promise rejected. Rejecting with a plain string does not preserve stack trace.

```
throw new Error("No deployment items to be deploy.");
```